inc/ChangeLog/RevisionInfo.php

```diff
@@ -115,146 +166,167 @@ public function editor()
      *
      * @return string
      */
-    public function itemName()
+    public function showFileName()
     {
-        $id = $this->info['id'];
-        $rev = ($this->info['current']) ? '' : $this->info['date'];
+        $id = $this->val('id');
+        $rev = $this->isCurrent() ? '' : $this->val('date');
 
-        switch ($this->info['item']) {
+        switch ($this->val('mode')) {
             case 'media': // media file revision
                 $params = ['tab_details'=> 'view', 'ns'=> getNS($id), 'image'=> $id];
                 if ($rev) $params += ['rev'=> $rev];
                 $href = media_managerURL($params, '&');
-                if(file_exists(mediaFN($id, $rev))) {
-                    $class = 'wikilink1';
-                } else {
-                    $class = 'wikilink2';
-                    if(!$this->info['current']) {
-                        //revision is not in attic
-                        return $id;
-                    }
-                }
-                return '<a href="'.$href.'" class="'.$class.'">'.$id.'</a>';
+                $display_name = $id;
+                $class = file_exists(mediaFN($id, $rev)) ? 'wikilink1' : 'wikilink2';
+                break;
             case 'page': // page revision
                 $params = $rev ? ['rev'=> $rev] : [];
                 $href = wl($id, $params, false, '&');
                 $display_name = useHeading('navigation') ? hsc(p_get_first_heading($id)) : $id;
                 if (!$display_name) $display_name = $id;
-                if(page_exists($id, $rev)) {
-                    $class = 'wikilink1';
-                } else {
-                    $class = 'wikilink2';
-                    if(!$this->info['current']) {
-                        //revision is not in attic
-                        return $display_name;
-                    }
-                }
-                return '<a href="'.$href.'" class="'.$class.'">'.$display_name.'</a>';
+                $class = page_exists($id, $rev) ? 'wikilink1' : 'wikilink2';
```

Comment:
```suggestion
if(page_exists($id, $rev)) {
    $class = 'wikilink1';
} else {
    if($this->isCurrent()) {
        //show only not-existing link for current page, which allows for directly create a new page/upload
        $class = 'wikilink2';
    } else {
        //revision is not in attic
        return $display_name;
    }
}
```

Reply:
and eventually move this piece of code to end of function and store state of revision `$isExisting = page_exists($id, $rev)` / `$isExisting = file_exists(mediaFN($id, $rev))` first?

Reply:
applied in 2aa019d